src/overlays/actors/ovl_En_Horse/z_en_horse.c

```diff
@@ -90,253 +139,3541 @@ static ColliderJntSphInit D_80A65F28 = {
+        this->flags |= ENHORSE_TURNING_TO_PLAYER;
+        this->angleToPlayer = Actor_WorldYawTowardActor(&this->actor, &PLAYER->actor);
+        angleDiff = (f32)this->angleToPlayer - (f32)this->actor.world.rot.y;
+        if (angleDiff > 32767.f) {
+            angleDiff -= 32767.0f;
+        } else if (angleDiff < -32767) {
+            angleDiff += 32767;
+        }
+
+        this->followPlayerTurnSpeed = angleDiff / Animation_GetLastFrame(sAnimationHeaders[this->type][this->animationIdx]);
+        Animation_PlayOnce(&this->skin.skelAnime, sAnimationHeaders[this->type][this->animationIdx]);
+        this->skin.skelAnime.playSpeed = 1.0f;
+        this->flags &= ~ENHORSE_LAND2_SOUND;
+        this->unk_21C = this->unk_228;
+    } else if (this->flags & ENHORSE_TURNING_TO_PLAYER) {
+        this->actor.world.rot.y = this->actor.world.rot.y + this->followPlayerTurnSpeed;
+        this->actor.shape.rot.y = this->actor.world.rot.y;
+        if (this->curFrame > 25.0f) {
+            if (!(this->flags & ENHORSE_LAND2_SOUND)) {
+                this->flags |= ENHORSE_LAND2_SOUND;
+                Audio_PlaySoundGeneral(NA_SE_EV_HORSE_LAND2, &this->actor.projectedPos, 4, &D_801333E0, &D_801333E0,
+                                       &D_801333E8);
+            }
+        }
+    } else {
+        EnHorse_RotateToPlayer(this, globalCtx);
+    }
+
+    if (this->animationIdx == ENHORSE_ANIM_GALLOP) {
+        this->actor.speedXZ = 8;
+        this->skin.skelAnime.playSpeed = this->actor.speedXZ * 0.3f;
+    } else if (this->animationIdx == ENHORSE_ANIM_TROT) {
+        this->actor.speedXZ = 6;
+        this->skin.skelAnime.playSpeed = this->actor.speedXZ * 0.375f;
+    } else if (this->animationIdx == ENHORSE_ANIM_WALK) {
+        this->actor.speedXZ = 3;
+        EnHorse_PlayWalkingSound(this);
+        this->skin.skelAnime.playSpeed = this->actor.speedXZ * 0.75f;
+    } else {
+        this->actor.speedXZ = 0;
+        this->skin.skelAnime.playSpeed = 1.0f;
+    }
+
+    if (!(this->flags & ENHORSE_TURNING_TO_PLAYER) && ++this->followTimer > 300) {
+        EnHorse_StartIdleRidable(this);
+        this->unk_21C = this->unk_228;
+
+        if (this->flags & ENHORSE_DRAW) {
+            Audio_PlaySoundGeneral(NA_SE_EV_HORSE_NEIGH, &this->unk_21C, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        }
+    }
+
+    if (SkelAnime_Update(&this->skin.skelAnime)) {
+        if (this->animationIdx == ENHORSE_ANIM_GALLOP) {
+            EnHorse_PlayGallopingSound(this);
+        } else if (this->animationIdx == ENHORSE_ANIM_TROT) {
+            EnHorse_PlayTrottingSound(this);
+        }
+        this->flags &= ~ENHORSE_TURNING_TO_PLAYER;
+        if (distToPlayer < 100.0f) {
+            EnHorse_StartIdleRidable(this);
+        } else {
+            EnHorse_SetFollowAnimation(this, globalCtx);
+        }
+    }
+}
+
+void EnHorse_UpdateIngoHorseAnim(EnHorse* this);
+void EnHorse_InitIngoHorse(EnHorse* this) {
```

Comment:
just a small thing, but throughout the file imo we can space out the forward declarations from the actual functions just to make it easier on the eyes